tests/test_datetime_utils.py

```diff
@@ -1,37 +1,111 @@
 
-    def test_query(self):
-        dt_today = datetime.date.today()
-        c_l_today = today()
-        self.assertNotEqual(c_l_today, None)
-        self.assertEqual(c_l_today.day, dt_today.day)
-        self.assertEqual(c_l_today.month, dt_today.month)
-        self.assertEqual(c_l_today.year, dt_today.year)
+    def _next_weekday(self, d, weekday):
+        days_ahead = weekday - d.weekday()
+        if days_ahead <= 0:  # Target day already happened this week
+            days_ahead += 7
+        return d + timedelta(days_ahead)
 
-        dt_yesterday = datetime.date.today() - timedelta(days=1)
+    def test_yesterday(self):
+        dt_yesterday = datetime.today() - timedelta(days=1)
         c_l_yesterday = yesterday()
         self.assertNotEqual(c_l_yesterday, None)
+        self.assertEqual(c_l_yesterday.minute, 0)
+        self.assertEqual(c_l_yesterday.hour, 0)
         self.assertEqual(c_l_yesterday.day, dt_yesterday.day)
         self.assertEqual(c_l_yesterday.month, dt_yesterday.month)
         self.assertEqual(c_l_yesterday.year, dt_yesterday.year)
 
-    def test_2(self):
-        self.assertNotEqual(sunday() + timedelta(days=-7), None)
-        self.assertNotEqual(sunday(), None)
+    def test_today(self):
+        dt_today = datetime.today()
+        c_l_today = today()
+        self.assertNotEqual(c_l_today, None)
+        self.assertEqual(c_l_today.minute, 0)
+        self.assertEqual(c_l_today.hour, 0)
+        self.assertEqual(c_l_today.day, dt_today.day)
+        self.assertEqual(c_l_today.month, dt_today.month)
+        self.assertEqual(c_l_today.year, dt_today.year)
 
-    def test_3(self):
-        self.assertNotEqual(month_begin(), None)
-        self.assertNotEqual(month_end(), None)
+    def test_tomorrow(self):
+        dt_tomorrow = datetime.today() - timedelta(days=-1)
+        c_l_tomorrow = tomorrow()
+        self.assertNotEqual(c_l_tomorrow, None)
+        self.assertEqual(c_l_tomorrow.minute, 0)
+        self.assertEqual(c_l_tomorrow.hour, 0)
+        self.assertEqual(c_l_tomorrow.day, dt_tomorrow.day)
+        self.assertEqual(c_l_tomorrow.month, dt_tomorrow.month)
+        self.assertEqual(c_l_tomorrow.year, dt_tomorrow.year)
 
-    def test_4(self):
+    def test_year(self):
         self.assertNotEqual(year_begin(), None)
+        self.assertEqual(year_begin(), datetime.utcnow().replace(month=1, day=1, hour=0, minute=0, second=0, microsecond=0))
         self.assertNotEqual(year_end(), None)
+        self.assertEqual(year_end(), datetime.utcnow().replace(year=datetime.utcnow().year + 1, month=1, day=1, hour=0, minute=0, second=0, microsecond=0))
+
+    def test_month(self):
+        self.assertNotEqual(month_begin(), None)
+        self.assertEqual(month_begin(), datetime.utcnow().replace(day=1, hour=0, minute=0, second=0, microsecond=0))
+        self.assertNotEqual(month_end(), None)
+        self.assertEqual(month_end(), (today().replace(day=1) + timedelta(days=32)).replace(day=1))
```

Comment:
this will cause minutes, seconds top be 0?

Reply:
yes, today() return minutes and seconds as 0, and it will simultaneously test the hour, minutes, and seconds too